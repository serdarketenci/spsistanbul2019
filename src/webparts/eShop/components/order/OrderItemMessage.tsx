import * as React from 'react';
import { mergeStyles } from 'office-ui-fabric-react/lib/Styling';
import { Icon } from 'office-ui-fabric-react/lib/Icon';

import { Label } from 'office-ui-fabric-react/lib/Label';


export interface IOrderItemMessageProps {
    message: string; 
    iconName: string; 
    iconColor: string;
  }

export default function OrderItemMessage(props:IOrderItemMessageProps) {
    const iconClass = mergeStyles({
        fontSize: 70,
        height: 50,
        width: 50,
        margin: '0 25px',
        color: props.iconColor
    });

    const containerClass = mergeStyles({
        textAlign:"center"
    });

    return (<div className={containerClass}>
        <Icon
            iconName={props.iconName}
            className={iconClass}
        />
        <Label>{props.message}</Label>
    </div>);
}
