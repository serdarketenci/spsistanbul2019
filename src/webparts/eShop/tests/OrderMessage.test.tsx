import React from "react";
import 'jest';
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import OrderItemMessage from '../components/order/OrderItemMessage';

describe('component renders properly', () => {

  const wrapper = shallow(<OrderItemMessage
    iconName="Accept"
    iconColor="#107c10"
    message="test2"
  />);

  test('snapshot should match', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});