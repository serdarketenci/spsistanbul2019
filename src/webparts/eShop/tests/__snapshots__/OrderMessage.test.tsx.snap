// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component renders properly snapshot should match 1`] = `
<div
  className="css-34"
>
  <StyledIconBase
    className="css-33"
    iconName="Accept"
  />
  <StyledLabelBase>
    test2
  </StyledLabelBase>
</div>
`;
